components/kserve/kserve.go

```diff
@@ -1,190 +0,0 @@
-		}
-
-		if strings.Contains(subcomponent.URI, ComponentName) {
-			// Download subcomponent
-			if err := deploy.DownloadManifests(ctx, ComponentName, subcomponent); err != nil {
-				return err
-			}
-			// If overlay is defined, update paths
-			defaultKustomizePath := "overlays/odh"
-			if subcomponent.SourcePath != "" {
-				defaultKustomizePath = subcomponent.SourcePath
-			}
-			Path = filepath.Join(deploy.DefaultManifestPath, ComponentName, defaultKustomizePath)
-		}
-	}
-
-	return nil
-}
-
-func (k *Kserve) GetComponentName() string {
-	return ComponentName
-}
-
-func (k *Kserve) ReconcileComponent(ctx context.Context, cli client.Client,
-	l logr.Logger, owner metav1.Object, dscispec *dsciv1.DSCInitializationSpec, platform cluster.Platform, _ bool) error {
-	enabled := k.GetManagementState() == operatorv1.Managed
-	monitoringEnabled := dscispec.Monitoring.ManagementState == operatorv1.Managed
-
-	if !enabled {
-		if err := k.removeServerlessFeatures(ctx, cli, owner, dscispec); err != nil {
-			return err
-		}
-	} else {
-		// Configure dependencies
-		if err := k.configureServerless(ctx, cli, l, owner, dscispec); err != nil {
-			return err
-		}
-		if k.DevFlags != nil {
-			// Download manifests and update paths
-			if err := k.OverrideManifests(ctx, platform); err != nil {
-				return err
-			}
-		}
-	}
-
-	if err := k.configureServiceMesh(ctx, cli, owner, dscispec); err != nil {
-		return fmt.Errorf("failed configuring service mesh while reconciling kserve component. cause: %w", err)
-	}
-
-	if err := deploy.DeployManifestsFromPath(ctx, cli, owner, Path, dscispec.ApplicationsNamespace, ComponentName, enabled); err != nil {
-		return fmt.Errorf("failed to apply manifests from %s : %w", Path, err)
-	}
-
-	l.WithValues("Path", Path).Info("apply manifests done for kserve")
-
-	if enabled {
-		if err := k.setupKserveConfig(ctx, cli, l, dscispec); err != nil {
-			return err
-		}
-
-		// For odh-model-controller
-		if err := cluster.UpdatePodSecurityRolebinding(ctx, cli, dscispec.ApplicationsNamespace, "odh-model-controller"); err != nil {
-			return err
-		}
-	}
-
-	if err := deploy.DeployManifestsFromPath(ctx, cli, owner, DependentPath, dscispec.ApplicationsNamespace, ComponentName, enabled); err != nil {
-		if !strings.Contains(err.Error(), "spec.selector") || !strings.Contains(err.Error(), "field is immutable") {
-			// explicitly ignore error if error contains keywords "spec.selector" and "field is immutable" and return all other error.
-			return err
-		}
-	}
-	l.WithValues("Path", Path).Info("apply manifests done for odh-model-controller")
-
-	// Wait for deployment available
-	if enabled {
-		if err := cluster.WaitForDeploymentAvailable(ctx, cli, ComponentName, dscispec.ApplicationsNamespace, 20, 3); err != nil {
-			return fmt.Errorf("deployment for %s is not ready to server: %w", ComponentName, err)
-		}
-	}
-
-	// CloudService Monitoring handling
-	if platform == cluster.ManagedRhods {
```

Comment:
@grdryn (Not a review but a question) How/ Where are you handling this functionality in the new refactor?